### PR TITLE
add idiom 259 Split on several separators

### PR DESCRIPTION
**Pull request by simonvonhackewitz**
Idiom 259: [Split on several separators](https://www.programming-idioms.org/idiom/259/split-on-several-separators)

### CODE REVIEW

File: src/idiom259.fuzion [hunk 1]
```diff
@@ -0,0 +1 @@
+parts := s.replace("_", ",").replace("-", ",").split(",")
```
Comment (fridis) [inline src/idiom259.fuzion:1]: Call syntax `(a b c)` is preferred over `a(b,c)`, so better use this
```suggestion
parts := s.replace "_"  ","
          .replace "-"  ","
          .split ","
```

File: src/idiom259ex.fz [hunk 2]
```diff
@@ -0,0 +1,5 @@
+ex259 is
+  strings := ["a-b_c,d", "-a--cc_,eee,ffff_", "---", ",_-", "abcd", ""]
+
+  _ := for s in strings do
+          say "\"$s\" : {s.replace("_", ",").replace("-", ",").split(",")}"
```
Comment (fridis) [inline src/idiom259ex.fz:5]: better use exactly the code from `idiom259.fuzion`, i.e., `parts := s.replace...`

**Review by fridis — CHANGES_REQUESTED** [review]

**Comment by michaellilltokiwa** [comment]
Please add a variant using `String.fields_func`

**Review by michaellilltokiwa — APPROVED** [review]

**Comment by michaellilltokiwa** [comment]
Nice

**Pull request merged**